src/tests/ftest/util/run_utils.py

```diff
@@ -314,22 +381,43 @@ def run_local(log, command, verbose=True, timeout=None, task_debug=False, stderr
         verbose (bool, optional): log the command output. Defaults to True.
         timeout (int, optional): number of seconds to wait for the command to complete.
             Defaults to None.
-        task_debug (bool, optional): whether to enable debug for the task object. Defaults to False.
         stderr (bool, optional): whether to enable stdout/stderr separation. Defaults to False.
+        capture_output (bool, optional): whether to include stdout/stderr in the CommandResult.
+            Defaults to True.
 
     Returns:
         CommandResult: groups of command results from the same hosts with the same return status
     """
     local_host = NodeSet(gethostname().split(".")[0])
-    task = task_self()
-    task.set_info('debug', task_debug)
-    task.set_default("stderr", stderr)
-    if verbose:
-        if timeout is None:
-            log.debug("Running on %s without a timeout: %s", local_host, command)
-        else:
-            log.debug("Running on %s with a %s second timeout: %s", local_host, timeout, command)
-    task.run(command=command, key=str(local_host), timeout=timeout)
+    kwargs = {
+        "encoding": "utf-8",
+        "shell": False,
+        "check": False,
+        "timeout": timeout,
+        "env": os.environ.copy()
+    }
+    if capture_output:
+        kwargs["stdout"] = subprocess.PIPE
+        kwargs["stderr"] = subprocess.PIPE if stderr else subprocess.STDOUT
```

Comment:
Below in this function:
```
    if verbose:
        results.log_output(log)
    else:
        # Always log any failed commands
        for data in results.output:
            if not data.passed:
                log_result_data(log, data)
```

We should probably skip any logging for `capture_output=False` because stdout and stderr will aready be printed. It would be misleading to print `<no output>` in this case

Reply:
Agreed, thanks.